src/main/java/com/redhat/digkins/services/TriggerBuildService.java

```diff
@@ -0,0 +1,116 @@
+package com.redhat.digkins.services;
+
+import com.offbytwo.jenkins.JenkinsServer;
+import com.offbytwo.jenkins.model.Executable;
+import com.offbytwo.jenkins.model.JobWithDetails;
+import com.offbytwo.jenkins.model.QueueItem;
+import com.offbytwo.jenkins.model.QueueReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Provides functionality to trigger a build.
+ **/
+public class TriggerBuildService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TriggerBuildService.class);
+
+  /**
+   * How long should we wait before we start checking the queue item status.
+   */
+  public static final long FIRST_CHECK_DELAY = 5 * 1000L;
```

Comment:
Just a suggestion: the time unit should be always explicit. In this case I believe they are millis so I would either name it `FIRST_CHECK_DELAY_MILLIS` or include it in comment `/** [...] to queue item status, in milli seconds. */`